### ISSUE TITLE
Issue with multiple dependencies when building parthenon externally

### ISSUE THREAD

**Issue by anku94** [parthenon-hpc-lab/parthenon#913]
(Edited to highlight issue with another dependency).

When building parthenon as a library, separately from say AthenaPK, it does not export some dependencies properly, and downstream codes can not find them. The dependencies are `std::filesystem` and `hdf5`.

1. For `std::filesystem`, it should either be exported via `cmake/parthenonConfig.cmake.in` (in which case its Find module will also have to be copied to the install tree), or the dependency can just be made `PRIVATE`.

        target_link_libraries(parthenon PRIVATE std::filesystem)
    
2. For HDF5, the build system defines and exports a proxy library for its C components (`add_library(HDF5_C ...)`). This only works if the variable `HDF5_C_LIBRARIES` points to a file path for the `.so` file. If the variable points to a CMake target (`hdf5-shared`), this does not work, as the target gets interpreted as a physical library.

    I think the portable way of doing it is just linking against the actual target (`hdf5::hdf5`), and adding `find_package(HDF5)` to `cmake/parthenonConfig.cmake.in`.

These changes worked for me on Ubuntu 20.04 with gcc-9 and CMake 3.27.

**Comment by pgrete** [comment]
Thanks for reporting.
Just so that I fully understand: This happens when you build Parthenon as a library and not as part of a downstream application code?
I didn't come across the issue yet so I want to make sure I understand (the issue and the fix).

PS: We've been discussing removing the capability to build Parthenon as a library as we weren't aware of anyone doing so (which means there's currently also no one testing this).

**Comment by anku94** [comment]
> Just so that I fully understand: This happens when you build Parthenon as a library and not as part of a downstream application code?

Yes - I found another minor bug in AthenaPK (linking with the target `parthenon` vs `Parthenon::parthenon`) so I figured that this is not something used/tested very often.

I'm not sure what the pros/cons are of deprecating library builds entirely, but I'm happy to keep testing this path and submit patches when I come across issues. At our lab, we use an "umbrella" system to build projects with complex dependencies - it's like Spack but less of a hassle, and having one Parthenon build and sharing it among multiple downstream codes works better.

https://github.com/anku94/amr-umbrella/tree/master/umbrella/umbrella

**Comment by anku94** [comment]
Updated description with issues I encountered with HDF5.

**Comment by pgrete** [comment]
Thanks for tracking down more issues.
Do you want to open a PR that includes the fixes?
